### PR TITLE
Allow client color changes to respect view tint

### PR DESCRIPTION
**Pull request by FlameArrow57**
[UI][QOL]

## About the PR 
This PR adds a flag to mobs that if set will allow any client color changes for that mob to only be set if the mob's view tint is turned on.

Allows Flock players to remove the teal tint from their screen

## Why's this needed? 
Was implemented for Flock, in case Flock players don't like the teal tint. 

Figured it could extend for other mobs if wanted

### CODE REVIEW

**Comment by FlameArrow57** [comment]
Requested for Flock awareness, up to you if you want to review

File: code/mob.dm [hunk 1]
```diff
@@ -1665,7 +1666,11 @@
 				else
 					color_matrix_2_apply = mult_color_matrix(color_matrix_2_apply, src.color_matrices[cmatrix])
 			src.active_color_matrix = color_matrix_2_apply
-	src.client?.set_color(src.active_color_matrix)
+	if (src.client)
+		if (!src.respect_view_tint_settings)
+			src.client.set_color(src.active_color_matrix)
+		else
+			src.client.set_color(src.client.view_tint ? src.active_color_matrix : null)
```
Comment (TobleroneSwordfish) [inline code/mob.dm:1673]:
```suggestion
	src.client?.set_color(src.active_color_matrix, src.respect_view_tint_settings)
```

File: code/mob.dm [hunk 2]
```diff
@@ -1646,11 +1647,11 @@
 	else
 		src.color_matrices -= label
 
-	src.update_active_matrix()
+	src.update_active_matrix(respect_view_tint_settings)
 
 /// Multiplies all of the mob's color matrices together and puts the result into src.active_color_matrix
 /// This matrix will be applied to the mob at the end of this proc, and any time the client logs in
-/mob/proc/update_active_matrix()
+/mob/proc/update_active_matrix(respect_view_tint_settings)
```
Comment (TobleroneSwordfish) [inline code/mob.dm:1654]: This argument doesn't appear to be used?

Comment (FlameArrow57) [reply code/mob.dm:1654]: It's used in line 1670 right now, or 1669 with your proposed changes

File: code/mob.dm (hunk 1 above)
Comment (FlameArrow57) [reply code/mob.dm:1673]: Won't work, if respect_view_tint_settings is true, then it will set your color even if you have your view tint off

**Review by pali6 — APPROVED** [review]
Sure looks good I think.

**Pull request merged**